specs/CustomScrollbar.md

```diff
@@ -0,0 +1,88 @@
+Custom Scrollbar 
+===
+
+# Background
+Developers would like to be able to set the scrollbar to Windows overlay style and match the Windows theme, but browser flags keep changing and are not reliable. Thus we want to provide an extendable API that's allow developers to define a scrollbar style in their app.
+
+# Examples
+## WinRT and .NET   
+```c#
+/// Create WebView Environment with option
+async void CreateEnvironmentWithOption()
+{
+    CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
+    options.CustomScrollbarStyle = COREWEBVIEW2_SCROLLBAR_WIN_FLUENT_OVERLAY_STYLE;
+    CoreWebView2Environment environment = await CoreWebView2Environment.CreateAsync(options: options);
+    webview.EnsureCoreWebView2Async(environment);
+}
+```
+
+## Win32 C++
+```cpp
+void AppWindow::InitializeWebView()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    COREWEBVIEW2_SCROLLBAR_STYLE style = COREWEBVIEW2_SCROLLBAR_WIN_FLUENT_OVERLAY_STYLE;
+    CHECK_FAILURE(options->SetCustomScrollbarStyle(styles));
+
+    // ... other option properties
+
+    // ... CreateCoreWebView2EnvironmentWithOptions
+}
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## Win32 C++
+
+```IDL
+interface ICoreWebView2EnvironmentOptions7;
+
+[v1_enum]
+typedef enum COREWEBVIEW2_SCROLLBAR_STYLE {
+  /// Browser default style
+  COREWEBVIEW2_SCROLLBAR_BROWSER_DEFAULT,
+  /// Window Style fluent overlay scroll bar
+  COREWEBVIEW2_SCROLLBAR_WIN_FLUENT_OVERLAY_STYLE
```

Comment:
Do we want to say 'Windows' here? Will this behave the same on Mac or on Mac it will use Mac-style scrollbars? 

Is the choice between:
* 'browser' scrollbars vs Windows fluent design
* Or is it 'browser' scrollbars vs whatever the current OS scrollbar design is and when you're on Win10+ the OS scrollbar design is Windows fluent design?

Reply:
The `COREWEBVIEW2_SCROLLBAR_WIN_FLUENT_OVERLAY_STYLE` only applies to windows design (which match the OS style). Should we generalize the naming to `COREWEBVIEW2_SCROLLBAR_OS_FLUENT_OVERLAY_STYLE`?

Reply:
I mean, can it work on Mac - is the implementation tied to Windows? If its tied to Windows, then we don't need a generic name and that enum value just won't exist for Mac.

Assuming this is Windows-only, I like FluentOverlayStyle. 
If its not Windows only, then WindowsFluentOverlayStyle sounds good.

In either case, please also add a link to what fluent UI is https://developer.microsoft.com/fluentui in the documentation for the enum.

Reply:
This is windows only. I will change it to FluentOverlayStyle then.

Reply:
Update to exclude windows keyword in enum